src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -2788,6 +2788,14 @@ internal PropertyCacheEntry(PropertyInfo property)
                 if (propertySetter != null && (propertySetter.IsPublic || propertySetter.IsFamily))
                 {
                     this.isStatic = propertySetter.IsStatic;
+                    foreach (var requiredCustomModifier in propertySetter.ReturnParameter.GetRequiredCustomModifiers())
+                    {
+                        if (!requiredCustomModifier.IsNested && requiredCustomModifier.Name == "IsExternalInit" && requiredCustomModifier.Namespace == "System.Runtime.CompilerServices")
```

Comment:
```suggestion
                        if (!requiredCustomModifier.IsNested && requiredCustomModifier.FullName == "System.Runtime.CompilerServices.IsExternalInit")
```

Slightly less work and string allocations